### PR TITLE
Use AVA for testing and test a few more scenarios

### LINKED ISSUE #26: Tests
Need some real tests for this. Could probably mock `process.stdout` and check the output.

### PR DESCRIPTION
**Pull request by djmitche**
Fixes #26.

I saw that ava was used in Ink so it seemed appropriate here, too.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -3,115 +3,140 @@ const test = require('ava');
...
 
-const withTerminal = opts => (t, testFn) => {
-	const terminal = new Terminal(opts);
+test('output a single line', t => {
+	const terminal = new Terminal({rows: 10, columns: 80});
 	terminal.state.setMode('crlf', true);
-
 	const log = logUpdate.create(terminal);
 
-	t.lineIs = (n, exp) => t.is(terminal.state.getLine(n).str, exp);
```
[the remark this answers is not in the document]
Comment (djmitche) [reply test.js]: xo wanted every method to only have a `t` argument, so it would have rejected that too :(

**Comment by djmitche** [comment]
I'll try to reproduce the Node-4 failure.

**Comment by djmitche** [comment]
Hm, `ava` seems to be the issue on node 4.  Even with an empty `test.js`, I get the same error:
```
  1 exception
  
  Uncaught exception in test.js
  undefined
  SyntaxError: [BABEL] /home/dustin/p/log-update/test.js: Unexpected token [ (While processing: "base$1")
  SyntaxError: [BABEL] /home/dustin/p/log-update/test.js: Unexpected token [ (While processing: "base$1")
      at exports.runInThisContext (vm.js:53:16)
      at Module._compile (module.js:373:25)
      at Object.Module._extensions..js (module.js:416:10)
      at Module.load (module.js:343:32)
      at Function.Module._load (module.js:300:12)
      at Module.require (module.js:353:17)
      at require (internal/module.js:12:17)
      at module.exports (/home/dustin/p/log-update/node_modules/ava/node_modules/@ava/babel-preset-transform-test-files/index.js:14:15)
      at /home/dustin/p/log-update/node_modules/ava/node_modules/hullabaloo-config-manager/build/helpers.js:93:59
      at /home/dustin/p/log-update/node_modules/ava/node_modules/@babel/core/lib/config/full.js:172:14
      at cachedFunction (/home/dustin/p/log-update/node_modules/ava/node_modules/@babel/core/lib/config/caching.js:42:17)
      at loadPresetDescriptor (/home/dustin/p/log-update/node_modules/ava/node_modules/@babel/core/lib/config/full.js:243:63)
      at /home/dustin/p/log-update/node_modules/ava/node_modules/@babel/core/lib/config/full.js:68:19
      at Array.map (native)
      at recurseDescriptors (/home/dustin/p/log-update/node_modules/ava/node_modules/@babel/core/lib/config/full.js:66:36)
      at loadFullConfig (/home/dustin/p/log-update/node_modules/ava/node_modules/@babel/core/lib/config/full.js:112:6)
  From previous event:
      at /home/dustin/p/log-update/node_modules/ava/api.js:128:83

shell returned 1
```

It looksl like the issue is that babel-plugin-throws-helper [doesn't support node 4](https://github.com/avajs/babel-plugin-throws-helper/blob/master/package.json) and [uses destructuring assignment](https://github.com/avajs/babel-plugin-throws-helper/blob/master/index.js#L32) which is not supported in node 4.

So, I'm not sure how to proceed..

File: test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply test.js]: In particular, the error from xo is "Test should only have one parameter named **t**".  I did move some of the boilerplate to a "setup" function though -- that makes it a little better-looking.

**Comment by sindresorhus** [comment]
Don't worry about it. I intend to drop Node.js 4 support.

**Comment by sindresorhus** [comment]
Thank you! :)

**Pull request merged**